src/Services/BlazorStaticService.cs

```diff
@@ -69,57 +71,28 @@ internal async Task GenerateStaticPages(string appUrl)
                 continue;
             }
 
-            var outFilePath = Path.Combine(options.OutputFolderPath, page.OutputFile);
+            var outFilePath = Path.Combine(options.OutputFolderPath, page.OutputFile.TrimStart('/'));
 
             string? directoryPath = Path.GetDirectoryName(outFilePath);
-            if (directoryPath != null)
+            if (!string.IsNullOrEmpty(directoryPath))
                 Directory.CreateDirectory(directoryPath);
+
             await File.WriteAllTextAsync(outFilePath, content);
         }
     }
+
     /// <summary>
-    /// Adds non-parametrized Razor pages to the list of pages to be generated as static content.
-    /// This method scans specified directories for Razor files and identifies those with a @page directive
-    /// that do not include parameters (i.e., no '{}' in the directive). For each matching file, it constructs
-    /// a URL based on the @page directive and determines a corresponding file path for the static HTML output.
-    /// The method assumes that each URL path should map to a folder structure with an 'index.html' file in it,
-    /// as specified in options.IndexPageHtml.
+    /// Registers razor pages that have no parameters to be generated as static pages.
     /// </summary>
-    /// <remarks>
-    /// This operation is controlled by the options.AddNonParametrizedRazorPages flag. Only Razor files in
-    /// directories specified in options.RazorPagesPaths are considered. The method employs regular expression
-    /// matching to identify suitable @page directives.
-    /// </remarks>
-    void AddNonParametrizedRazorPages()
+    private void RegisterNonTokenizedPages()
     {
-        if (!options.AddNonParametrizedRazorPages) return;
-
-        var regex = new Regex("""
-                              @page "/(?!.*\{.*\})(.*?)"
-                              """);// Regular expression to match @page directive, but ignore when {} are present
+        var entryAssembly = Assembly.GetEntryAssembly()!;
```

Comment:
null shouldn't be an issue here because [it can only happen if the library is called from unmanaged code](<https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assembly.getentryassembly?view=net-8.0>)